core/types.ml

```diff
@@ -124,6 +124,61 @@ type ('t, 'r) session_type_basis =
     | `End ]
       [@@deriving show]
 
+type lens_delete_mode =
+  [ `DeleteLeft
+  | `DeleteRight
+  | `DeleteBoth ]
+      [@@deriving show]
+
+type lens_phrase = 
+  [ `Constant  of Constant.constant
+  | `Var       of Operators.name 
+  | `InfixAppl of Operators.binop * lens_phrase * lens_phrase
+  | `UnaryAppl of Operators.unary_op * lens_phrase 
+  (* the In operator checks if the tuple generated by the names list is 
+   * in the list of constant tuples *)
+  | `In        of Operators.name list * (Constant.constant list) list
+  | `Case      of lens_phrase option * (lens_phrase * lens_phrase) list * lens_phrase
+  | `TupleLit  of lens_phrase list
+  ]
+      [@@deriving show]
+
+(* Lenses *)
+module ColSet = struct
+    include StringSet
+
+    let pp_pretty fmt cs = 
+        List.iter (fun c -> Format.pp_print_string fmt c; Format.pp_print_space fmt ()) (elements cs)
+end
+
+type colset = ColSet.t
+      [@@deriving show]
+
+module FunDep = struct 
```

Comment:
Relevance to types? Looks like something that is utility related.

Reply:
Moved into LensUtility.